src/Repository/PostRepository.php

```diff
@@ -36,31 +35,24 @@ public function __construct(ManagerRegistry $registry)
         parent::__construct($registry, Post::class);
     }
 
-    public function findLatest(int $page = 1, Tag $tag = null): Pagerfanta
+    public function findLatest(int $page = 1, Tag $tag = null): Paginator
     {
         $qb = $this->createQueryBuilder('p')
             ->addSelect('a', 't')
             ->innerJoin('p.author', 'a')
             ->leftJoin('p.tags', 't')
             ->where('p.publishedAt <= :now')
             ->orderBy('p.publishedAt', 'DESC')
-            ->setParameter('now', new \DateTime());
+            ->setParameter('now', new \DateTime())
+            ->setFirstResult(($page - 1) * Post::NUM_ITEMS)
```

Comment:
you no longer deal with negative page numbers properly here (Pagerfanta was validating that).